libraries/botbuilder-dialogs-adaptive-testing/src/testActions/userActivity.ts

```diff
@@ -61,7 +61,10 @@ export class UserActivity extends TestAction implements UserActivityConfiguratio
             activity.from = Object.assign({}, activity.from);
             activity.from.id = this.user;
             activity.from.name = this.user;
+        } else if (this.activity?.from && Object.keys(this.activity.from)) {
```

Comment:
Is this meant to check if `activity.from` is an object? If so, check out the new [`botbuilder-stdlib`](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-stdlib/src/types.ts#L539) which has some type tests and assertions. In particular this could be:

```typescript```
import { tests } from 'botbuilder-stdlib';
...
} else if (tests.isObject(this.activity?.from)) {
  ...
}
```

Reply:
It's meant to check that it isn't an empty object. Open to suggestions.

Reply:
`Object.keys` returns `[]` for an empty object, and `if ([]) { console.log('foo') }` does, indeed, log `foo`. I'd suggest perhaps using:

```typescript
import isEmpty from 'lodash/isEmpty'

...

else if (!isEmpty(this.activity?.from)) {
  ...
}
```

This better conveys what you are trying to verify.

Reply:
I ended up using your first example, since I think it makes sense to allow for an empty `from` property